certificate-service/app/src/main/java/se/inera/intyg/certificateservice/application/certificate/service/validation/RevokeCertificateRequestValidator.java

```diff
@@ -0,0 +1,23 @@
+package se.inera.intyg.certificateservice.application.certificate.service.validation;
+
+import static se.inera.intyg.certificateservice.application.common.validator.ValidationUtil.validateCertificateId;
+import static se.inera.intyg.certificateservice.application.common.validator.ValidationUtil.validateRevokeReason;
+import static se.inera.intyg.certificateservice.application.common.validator.ValidationUtil.validateUnit;
+import static se.inera.intyg.certificateservice.application.common.validator.ValidationUtil.validateUnitExtended;
+import static se.inera.intyg.certificateservice.application.common.validator.ValidationUtil.validateUser;
+
+import org.springframework.stereotype.Component;
+import se.inera.intyg.certificateservice.application.certificate.dto.RevokeCertificateRequest;
+
+@Component
+public class RevokeCertificateRequestValidator {
+
+  public void validate(RevokeCertificateRequest request, String certificateId) {
+    validateUser(request.getUser());
+    validateUnitExtended(request.getUnit(), "Unit");
+    validateUnit(request.getCareUnit(), "CareUnit");
+    validateUnit(request.getCareProvider(), "CareProvider");
+    validateCertificateId(certificateId);
+    validateRevokeReason(request.getRevoked().getReason());
```

Comment:
It looks like the reason should always be set and if it is of a specific type of reason (ANNAT_ALLVARLIGT_FEL) then it is mandatory to give a message.